sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -2169,12 +2583,53 @@ public void query(OutputStream stream, String expression) {
     public BlobQueryResponse queryWithResponse(BlobQueryOptions queryOptions, Duration timeout, Context context) {
         StorageImplUtils.assertNotNull("options", queryOptions);
         StorageImplUtils.assertNotNull("outputStream", queryOptions.getOutputStream());
-        Mono<BlobQueryResponse> download = client
-            .queryWithResponse(queryOptions, context)
-            .flatMap(response -> FluxUtil.writeToOutputStream(response.getValue(), queryOptions.getOutputStream())
-                .thenReturn(new BlobQueryResponse(response)));
+        Context finalContext = context == null ? Context.NONE : context;
+        BlobRequestConditions requestConditions = queryOptions.getRequestConditions() == null
+            ? new BlobRequestConditions() : queryOptions.getRequestConditions();
+        QuerySerialization in = BlobQueryReader.transformInputSerialization(queryOptions.getInputSerialization(),
+            LOGGER);
+        QuerySerialization out = BlobQueryReader.transformOutputSerialization(queryOptions.getOutputSerialization(),
+            LOGGER);
+
+        QueryRequest qr = new QueryRequest()
+            .setExpression(queryOptions.getExpression())
+            .setInputSerialization(in)
+            .setOutputSerialization(out);
+
+        Callable<ResponseBase<BlobsQueryHeaders, InputStream>> operation = () ->
+            this.azureBlobStorage.getBlobs().queryWithResponse(containerName, blobName, getSnapshotId(), null,
+                requestConditions.getLeaseId(), requestConditions.getIfModifiedSince(),
+                requestConditions.getIfUnmodifiedSince(), requestConditions.getIfMatch(),
+                requestConditions.getIfNoneMatch(), requestConditions.getTagsConditions(), null, qr,
+                getCustomerProvidedKey(), finalContext);
+        try {
+            ResponseBase<BlobsQueryHeaders, InputStream> response = sendRequest(operation, timeout,
```

Comment:
This changed the scope of the `timeout`. Before it included writing the response to the `OutputStream`, it no longer includes that.